source/components/molecules/SwitchAction/SwitchAction.js

```diff
@@ -0,0 +1,96 @@
+/* eslint-disable no-nested-ternary */
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components/native';
+import { Button, Text } from 'source/components/atoms';
+import EditableList from '../EditableList';
+
+const InputWrapper = styled.View`
+  padding-bottom: 16px;
+`;
+
+const InpulList = props => (
+  <InputWrapper>
+    <EditableList {...props} />
+  </InputWrapper>
+);
+
+const SwitchAction = props => {
+  const { fields, theme } = props;
+  const [isAddressEditable, setisAddressEditable] = React.useState(false);
+  const [isPersonalInfoEditable, setisPersonalInfoEditable] = React.useState(false);
+
+  const addressButton = () => (
+    <Button z={0} size="small" onClick={() => setisAddressEditable(true)} color="blue">
+      <Text>Ändra</Text>
+    </Button>
+  );
+
+  const personalInfoButton = () => (
+    <Button z={0} size="small" onClick={() => setisPersonalInfoEditable(true)} color="blue">
+      <Text>Ändra</Text>
+    </Button>
+  );
+
+  // TODO: update data
+  const onInputChange = () => {};
+
+  const ComponentOptions =
+    fields && fields.length
```

Comment:
You do not need to pass all the fields down, since the mapping of fields is allready dealt with in the Step component. So this component only needs to care about rendering one field correctly

Reply:
There was no mapping of field done till now in Step component. I think the Step can simply pass all the field data to this component to handle.